task/send-to-webhook-discord/0.1/send-to-webhook-discord.yaml

```diff
@@ -0,0 +1,55 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: send-to-webhook-discord
+  labels:
+    app.kubernetes.io/version: "0.1"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/categories: Messaging
+    tekton.dev/tags: messaging
+    tekton.dev/displayName: "Send a message to Discord channel"
+    tekton.dev/platforms: "linux/amd64,linux/s390x,linux/ppc64le"
+spec:
+  description: >-
+    These tasks post a message to a Discord channel.
+
+    This task uses Webhooks integration of Discord channel to send a message.
+
+  params:
+  - name: webhook-secret
+    type: string
+    description: secret name of Discord webhook URL
+  - name: message
+    type: string
+    description: plain text message
+    default: 'Hello World from Tekton Bot'
+  - name: bot-name
+    type: string
+    description: plain text message
+    default: 'Tekton Bot'
+  - name: avatar-url
+    type: string
+    description: url avatar image
+    default: 'https://github.com/tektoncd.png'
+  steps:
+  - name: post
+    image: docker.io/curlimages/curl:7.70.0@sha256:031df77a11e5edded840bc761a845eab6e3c2edee22669fb8ad6d59484b6a1c4 #tag: 7.70.0
+    script: |
+      #!/usr/bin/env sh
+      MESSAGE=$(echo "${MESSAGE}" | sed -e 's/\"/\\\\"/g')
```

Comment:
quoting may be potentially be buggy still :\ 

i don't think there is a solution for that unless we use another programming language than shell 

but sounds good for now since there is not much alternatively

Reply:
can you give an example or suggestion to solve such a case @chmouel ?
i inspire to contribute because there is already similar solution like the task of webhook slack

https://github.com/tektoncd/catalog/blob/cebb548d561259fef7239f9c1e56db9b241c16f0/task/send-to-webhook-slack/0.1/send-to-webhook-slack.yaml#L35-L39

Reply:
Yep there is not many ways around this,